File: adapter-pattern/java/src/Mp4Player.java
package src;

// HlsPlayer播放器实现了视频接口
public class Mp4Player implements MediaPlayer {
  private String fileType;
  private String filePath;

  public Mp4Player(String fileType, String filePath) {
    this.fileType = fileType;
    this.filePath = filePath;
  }

  public void play() {
    System.out.println("Mp4Player fileType:" + fileType + " | filePath:" + filePath);
    System.out.println("Mp4Player::play()");
  }

  public void setFileType(String fileType) {
    this.fileType = fileType;
  }

  public void setFilePath(String filePath) {
    this.filePath = filePath;
  }
}

File: adapter-pattern/ts/src/Mp4Player.ts
import { MediaPlayer } from './MediaPlayer.js'
export class Mp4Player implements MediaPlayer {
  fileType: string;
  filePath: string;

  constructor(fileType: string, filePath: string) {
    this.fileType = fileType
    this.filePath = filePath
  }

  play() {
    console.log(
      'Mp4Player fileType:' + this.fileType + ' | filePath:' + this.filePath,
    )
    console.log('Mp4Player::play()')
  }
  setFileType(fileType: string) {
    this.fileType = fileType;
  }

  setFilePath(filePath: string) {
    this.filePath = filePath;
  }
}
